SimTracker/TrackTriggerAssociation/plugins/TTClusterAssociator.cc

```diff
@@ -21,32 +21,29 @@ void TTClusterAssociator<Ref_Phase2TrackerDigi_>::produce(edm::Event& iEvent, co
 
   /// Get the TrackingParticles
 
-  iEvent.getByToken(tpToken, TrackingParticleHandle);
+  iEvent.getByToken(tpToken, trackingParticleHandle);
 
   //  const TrackerTopology* const tTopo = theTrackerTopology.product();
   const TrackerGeometry* const theTrackerGeom = theTrackerGeometry.product();
 
   /// Preliminary task: map SimTracks by TrackingParticle
   /// Prepare the map
-  std::map<std::pair<unsigned int, EncodedEventId>, edm::Ptr<TrackingParticle>> simTrackUniqueToTPMap;
+  std::map<std::pair<unsigned int, EncodedEventId>, TrackingParticlePtr> simTrackUniqueToTPMap;
   simTrackUniqueToTPMap.clear();
 
-  if (!TrackingParticleHandle->empty()) {
+  if (not trackingParticleHandle->empty()) {
     /// Loop over TrackingParticles
-    unsigned int tpCnt = 0;
-    std::vector<TrackingParticle>::const_iterator iterTPart;
-    for (iterTPart = TrackingParticleHandle->begin(); iterTPart != TrackingParticleHandle->end(); ++iterTPart) {
+    for (unsigned int tpCnt = 0; tpCnt < trackingParticleHandle->size(); tpCnt++) {
       /// Make the pointer to the TrackingParticle
-      edm::Ptr<TrackingParticle> tempTPPtr(TrackingParticleHandle, tpCnt++);
+      TrackingParticlePtr tempTPPtr(trackingParticleHandle, tpCnt);
 
       /// Get the EncodedEventId
       EncodedEventId eventId = EncodedEventId(tempTPPtr->eventId());
 
       /// Loop over SimTracks inside TrackingParticle
-      std::vector<SimTrack>::const_iterator iterSimTrack;
-      for (iterSimTrack = tempTPPtr->g4Tracks().begin(); iterSimTrack != tempTPPtr->g4Tracks().end(); ++iterSimTrack) {
+      for (const auto& simTrack : tempTPPtr->g4Tracks()) {
         /// Build the unique SimTrack Id (which is SimTrack ID + EncodedEventId)
-        std::pair<unsigned int, EncodedEventId> simTrackUniqueId(iterSimTrack->trackId(), eventId);
+        std::pair<unsigned int, EncodedEventId> simTrackUniqueId(simTrack.trackId(), eventId);
         simTrackUniqueToTPMap.insert(std::make_pair(simTrackUniqueId, tempTPPtr));
```

Comment:
this and the previous line can be simplified to:
```cpp
simTrackUniqueToTPMap.emplace(std::make_pair(simTrack.trackId(), eventId), tempTPPtr);
```

Reply:
done